pkg/client/rootd/dbus/resolved.go

```diff
@@ -94,8 +97,20 @@ func SetLinkDomains(c context.Context, networkIndex int, domains ...string) erro
 			}
 			dds = append(dds, resolvedDomain{Name: domain, RoutingOnly: routing})
 		}
-		return conn.Object("org.freedesktop.resolve1", "/org/freedesktop/resolve1").CallWithContext(
-			c, "org.freedesktop.resolve1.Manager.SetLinkDomains", 0, int32(networkIndex), dds).Err
+		for i := 0; i < len(dds); i += maxDomainsPerBatch {
+			end := i + maxDomainsPerBatch
+			if end > len(dds) {
+				end = len(dds)
+			}
+			if err := conn.Object("org.freedesktop.resolve1", "/org/freedesktop/resolve1").CallWithContext(
+				c, "org.freedesktop.resolve1.Manager.SetLinkDomains", 0, int32(networkIndex), dds[i:end]).Err; err != nil {
```

Comment:
This implementation suggests that `SetLinkDomains` will append the domains to an existing set. I'm under the impression that it overwrites any previous setting, and the [documentation](https://www.freedesktop.org/software/systemd/man/latest/org.freedesktop.resolve1.html) does not mention anything about appending. If I'm right, then this implementation will just set the last batch and throw away any previous batches.

Do you have any evidence proving that an append happens here?